cockatrice/src/pictureloader.cpp

```diff
@@ -499,21 +501,40 @@ void PictureLoaderWorker::picDownloadFinished(QNetworkReply *reply)
     imgReader.setDecideFormatFromContent(true);
     imgReader.setDevice(reply);
 
-    if (imgReader.read(&testImage)) {
+    bool logSuccessMessage = false;
+
+    static const int riffHeaderSize = 12; // RIFF_HEADER_SIZE from webp/format_constants.h
+    auto replyHeader = reply->peek(riffHeaderSize);
+
+    if (replyHeader.startsWith("RIFF") && replyHeader.endsWith("WEBP")) {
+        auto imgBuf = QBuffer(this);
+        imgBuf.setData(reply->readAll());
+
+        auto movie = QMovie(&imgBuf);
```

Comment:
why does qimagereader not work? can we give it a format hint?

Reply:
Apparently some Webps are actually "movies" and not images, which is weird. See [QTBUG-75044](https://bugreports.qt.io/browse/QTBUG-75044)